test/ng/directive/ngModelOptionsSpec.js

```diff
@@ -480,15 +480,37 @@ describe('ngModelOptions', function() {
         expect($rootScope.checkbox).toBeUndefined();
         $timeout.flush(4000);
         expect($rootScope.name).toEqual('a');
+
         helper.changeInputValueTo('b');
         browserTrigger(inputElm, 'blur');
         $timeout.flush(4000);
         expect($rootScope.name).toEqual('a');
         $timeout.flush(2000);
         expect($rootScope.name).toEqual('b');
+
+        helper.changeInputValueTo('c');
+        browserTrigger(helper.inputElm, 'mouseup');
+        // counter-intuitively `default` in `debounce` is a catch-all
+        expect($rootScope.name).toEqual('b');
+        $timeout.flush(10000);
+        expect($rootScope.name).toEqual('c');
       });
 
 
+      it('should trigger immediately for the default event if not listed in the debounce list',
```

Comment:
This is not specific to the `default` event. The bug affected any event that was not listed in `debounce` as long as `debounce` was an object and didn't contain the catch-all `default`. I wouldn't mind adding one more assertion to the test, like this:

```diff
          var inputElm = helper.compileInput(
               '<input type="text" ng-model="name" name="alias" ' +
                 'ng-model-options="{' +
-                  'updateOn: \'default blur\', ' +
+                  'updateOn: \'default blur foo\', ' +
                   'debounce: { blur: 5000 }' +
                 '}"' +
               '/>');
   
           helper.changeInputValueTo('a');
           expect($rootScope.name).toEqual('a');
+          helper.changeInputValueTo('b');
+          browserTrigger(inputElm, 'foo');
+          expect($rootScope.name).toEqual('b');
        });
```